src/usbg_schemes_libconfig.c

```diff
@@ -662,7 +699,7 @@ static int usbg_export_gadget_prep(usbg_gadget *g, config_setting_t *root)
 	}
 
 	node = config_setting_add(root, USBG_CONFIGS_TAG,
-				     CONFIG_TYPE_LIST);
```

Comment:
I'm not to happy with changing this.

To be honest I think that we don't need this as configs in contrast to functions can be easily identified using only the id.

Reply:
Hm, ok, assuming ID is unique will even simplify the code quite a bit. usbg_get_config seems to support searching by id only. So I convert the config property to be an integer only value, probably also rename it to be clear it references the id, e.g.:

```
os_descs : 
{
    config_id = 1;
    use = 1;
    qw_sign = "M";
    b_vendor_code = 0x5;
};
configs = (
    {   
        id = 1;
        name = "c";
...
```